fixtures/pages-workerjs-directory/tests/index.test.ts

```diff
@@ -33,9 +33,15 @@ describe("Pages _worker.js/ directory", () => {
 		await expect(
 			fetch(`http://${ip}:${port}/static`).then((resp) => resp.text())
 		).resolves.toContain("static");
+		await expect(
+			fetch(`http://${ip}:${port}/other-static`).then((resp) => resp.text())
+		).resolves.toContain("other-static");
```

Comment:
this shows a potential issue here, in this case we have `static.js` reachable via `/static` and `other-static.mjs` reachable via `/other-static`

we'd have a conflict if we had `static.js` and `static.mjs`, I imagine one of the two will win, is this an acceptable situation?

Reply:
What happens when a user has both `foo.js` and `foo.mjs` - Does it silently accept it?

Reply:
yeah, i tried with `static.js` and `static.mjs`, it seems like it simply returns the `js` one if you don't specify the extension:
![Screenshot 2023-08-04 at 17 26 33](https://github.com/cloudflare/workers-sdk/assets/61631103/a91b3b78-f967-414d-a63f-87120a02ab51)

Reply:
What do these test actually mean? If the files are inside the `_worker.js` directory then they should get bundled up into the Worker and no made available as static assets. Is this just a bug in `wrangler pages dev` or am I missing something?

Reply:
@petebacondarwin yeah sorry this is totally my bad...

I added the `other-static` test following what greg did for the `static` one, but after double checking now I noticed that he was actually testing this:
https://github.com/cloudflare/workers-sdk/blob/69dc41a17864f14d5d0ad8ed9cc86dc25adc1d8a/fixtures/pages-workerjs-directory/public/_worker.js/index.js#L13-L15

not the fact that we are accessing the javascript files directly, so I'm actually testing the wrong thing and need to add a `other-static` handle in the worker script 🤦 

I'll amend this! thanks a lot for noticing it! 🙏 

____

Regarding accessing the javascript files directly... that does look like a bug to me... probably something we should look into

Reply:
I've amended the tests here: https://github.com/cloudflare/workers-sdk/pull/3707/commits/697f8db8f114757357354ef443fa9a60307767cb

I've added some extra test that I think makes it clearer that what we're not testing the js files directly but their use in the worker